src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,402 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Finds and return if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        boolean contains = false;
+        for (DetailNode node : javadoc.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = node.getChildren();
+
+                // Checking size of child is not required, since child contains
+                // children of Inline Tag, as at least 2 children will be present
+                // which are RCURLY and LCURLY.
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                        && SUMMARY_TEXT.equals(child[1].getText())) {
+                    contains = true;
+                }
+            }
+        }
+
+        if (containsSummaryTagWithHtmlTag(javadoc)) {
```

Comment:
this method is hard to understand .... looks like you try to cover cases when Summary tag is placed inside html tags.
it is not very different from situation when no html tags at all.

You just need util method that find first JAVADOC_INLINE_TAG from root of javadoc, no mater how much html tags wraps it [0..many].

Reply:
if some common util method does not work for you, you just need to write your own util method that serve needs on this Check.

Reply:
Done.